WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginEmailFragment.java

```diff
@@ -0,0 +1,397 @@
+import android.view.View;
+import android.view.View.OnClickListener;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import org.greenrobot.eventbus.Subscribe;
+import org.greenrobot.eventbus.ThreadMode;
+import org.wordpress.android.BuildConfig;
+import org.wordpress.android.R;
+import org.wordpress.android.WordPress;
+import org.wordpress.android.fluxc.Dispatcher;
+import org.wordpress.android.fluxc.generated.AccountActionBuilder;
+import org.wordpress.android.fluxc.store.AccountStore.OnAvailabilityChecked;
+import org.wordpress.android.ui.accounts.AbstractFragment;
+import org.wordpress.android.ui.accounts.SignInDialogFragment;
+import org.wordpress.android.ui.accounts.login.nav.LoginNav;
+import org.wordpress.android.ui.accounts.login.nav.LoginStateGetter;
+import org.wordpress.android.util.AppLog;
+import org.wordpress.android.util.AppLog.T;
+import org.wordpress.android.util.EditTextUtils;
+import org.wordpress.android.util.NetworkUtils;
+import org.wordpress.emailchecker2.EmailChecker;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.inject.Inject;
+
+public class LoginEmailFragment extends AbstractFragment implements TextWatcher {
+    private static final String KEY_IN_PROGRESS = "KEY_IN_PROGRESS";
+
+    public static final String TAG = "login_email_fragment_tag";
+    public static final int MAX_EMAIL_LENGTH = 100;
+
+    protected TextInputLayout mEmailEditTextLayout;
+    protected EditText mEmailEditText;
+
+    protected boolean mEmailAutoCorrected;
+
+    protected Button mNextButton;
+    protected View mUsernamePasswordButton;
+
+    protected @Inject Dispatcher mDispatcher;
+
+    private LoginNav.InputEmail mLoginNavInputEmail;
+
+    private ProgressDialog mProgressDialog;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        ((WordPress) getActivity().getApplication()).component().inject(this);
+
+        setHasOptionsMenu(true);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        ViewGroup rootView = (ViewGroup) inflater.inflate(R.layout.login_email_screen, container, false);
+
+        mEmailEditText = (EditText) rootView.findViewById(R.id.login_email);
+        mEmailEditText.addTextChangedListener(this);
+
+        mEmailEditTextLayout = (TextInputLayout) rootView.findViewById(R.id.login_email_layout);
+
+        mNextButton = (Button) rootView.findViewById(R.id.login_email_next_button);
+        mNextButton.setOnClickListener(mNextClickListener);
+        mUsernamePasswordButton = rootView.findViewById(R.id.login_email_username_password);
+
+        mEmailEditText.setOnFocusChangeListener(new View.OnFocusChangeListener() {
+            public void onFocusChange(View v, boolean hasFocus) {
+                if (!hasFocus) {
```

Comment:
Thanks! Having used them now, I think I'm in favor of eliminating the automatic corrections.

I think the idea of suggesting a correction is good though. We've talked about using error messages to provide more helpful messages to the user. For example, if we detect an invalid email address, the error could include something like "Did you mean a@bt.com?" This could be something that comes after this initial implementation of the new design.